python/ray/data/_internal/pipeline_executor.py

```diff
@@ -49,10 +43,23 @@ def __init__(self, pipeline: "DatasetPipeline[T]"):
         else:
             self._bars = None
 
+        # The creation of execution thread pool is deferred until this is
```

Comment:
Could we also add an explicit error thrown if you try to serialize a PipelineExecutor that has already started? I think you can do this by defining `__reduce__`

Reply:
Done.